app/game/threat.js

```diff
@@ -2,23 +2,46 @@ const THREAT_TTL = 30;
 const THREAT_TICK_SPEED = 1000;
 const THREAT_COOLDOWN_SECONDS = 5;
 
-const Threat = (threatType, onThreatUnresolved) => {
+const THREAT_RESOLVE_TIME_MS = 3*1000; // How long it takes when a player enters a room w/ the right tool for the threat to disappear
+
+const Threat = (threatType, onThreatUnresolved, onThreatResolved) => {
     let currentAge = THREAT_TTL;
     const THREAT_TYPE = threatType;
+    let resolved = false;
+    let isResolving = false;
+    const MAP_THREAT_TO_TOOL = {
+        "fire" : "fire-extinguisher",
+        "breach" : "wrench",
+        "invader" : "gun"
+    }
 
     const tick = () => {
         currentAge -= 1;
```

Comment:
Remind me to talk to you irl about this. In short, I think we ought to be passing in a `deltaSeconds` to each of these tick functions instead of assuming 1 second has passed since the last time it was called